core/arch/arm/kernel/boot.c

```diff
@@ -188,6 +188,59 @@ static void init_vfp_nsec(void)
 }
 #endif
 
+static void check_crypto_extensions(void)
+{
+	bool ce_supported = true;
+
+	if (!feat_aes_implemented() &&
+	    IS_ENABLED(CFG_CRYPTO_AES_ARM_CE)) {
+		EMSG("AES instructions are not supported");
+		ce_supported = false;
+	}
```

Comment:
How about:
```C
	if (!feat_aes_implemented() &&
	    IS_ENABLED(CFG_CRYPTO_AES_ARM_CE))
		panic("AES instructions are not supported");
```
...and same below. In addition to making the code shorter it would print the exact feature that is missing (well, the first found at least).

Reply:
That was my first thought, but then I decided why not to print all of them?
Easier to tackle the issue, as you report all unsupported instructions at once

Reply:
Fair enough. Good with me.